tests/php/ORM/UnionListTest.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+namespace SilverStripe\ORM\Tests;
+
+use SilverStripe\Dev\Debug;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\UnionList;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\ORM\Tests\DataObjectTest\ValidatedObject;
+
+class UnionListTest extends SapphireTest
+{
+    protected static $fixture_file = 'UnionListTest.yml';
+
+    /**
+     * @var UnionList
+     */
+    protected $unionList = null;
+
+    public function setUp()
+    {
+        parent::setUp();
+        
+        // create an object to test with
+        /*$obj1 = new ValidatedObject();
+        $obj1->Name = 'test obj 1';
+        $obj1->write();
+        $this->assertTrue($obj1->isInDB());
+
+        $obj2 = new ValidatedObject();
+        $obj2->Name = 'test obj 2';
+        $obj2->write();
+        $this->assertTrue($obj2->isInDB());
+
+        $obj3 = new ValidatedObject();
+        $obj3->Name = 'test obj 3';
+        $obj3->write();
+        $this->assertTrue($obj3->isInDB());
+
+        $obj4 = new ValidatedObject();
+        $obj4->Name = 'test obj 4';
+        $obj4->write();
+        $this->assertTrue($obj4->isInDB());*/
+        $obj3 = ValidatedObject::get()->find('Name', 'test obj 3');
```

Comment:
you can use `$this->objFromFixtureValidatedObject::class, 'numeric1')` - this makes the tests a little more versatile.